### PR TITLE
Ignore fdbserver processes in check compatibility step

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

Fixes: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1468

## Type of change

*Please select one of the options below.*

- Bug fix (non-breaking change which fixes an issue)

## Discussion

Like discussed in the issue, the risk with this approach is that we are not catching incompatible processes that are running in the same Pod as a fdbserver process. I think the risk is minimal and in general we are not recommending to run any client workload colocated in the same Pod.

## Testing

Unit tests + e2e tests for upgrades from 7.1.25 -> 7.2.0

## Documentation

We could document the limitation and that we are not recommending to run any client workload in the same Pods.

## Follow-up

-

### CODE REVIEW

**Comment by sbodagala** [comment]
> We could document the limitation and that we are not recommending to run any client workload in the same Pods.

Maybe we need to understand why fdbserver processes are now getting reported as clients before we modify the operator to work with it (considering the limitation that it will impose)? (I see that nobody replied to your question on Slack, probably we should post this in the other broader channel.)

**Review by johscheuer — COMMENTED** [review]
Asked on the FDB Slack channel why fdbserver processes are added as clients and if there's a better way to filter them.

File: internal/foundationdb_status.go [hunk 1]
```diff
@@ -79,3 +81,50 @@ func GetMinimumUptimeAndAddressMap(cluster *fdbv1beta2.FoundationDBCluster, stat
...
+	}
+
+	var unsupportedClients []string
+	for _, versionInfo := range status.Cluster.Clients.SupportedVersions {
+		if versionInfo.ProtocolVersion == "Unknown" {
+			continue
+		}
+
+		if versionInfo.ProtocolVersion != protocolVersion {
+			for _, client := range versionInfo.MaxProtocolClients {
```
Comment (brownleej) [inline internal/foundationdb_status.go:114]: My understanding is that `max_protocol_clients` is a sample of the clients that have this version as their max supported protocol. Does that mean it's possible that all of the clients in the sample will be fdbserver processes, but we will have normal clients that are not included in the sample, leading to false negatives from this check?

**Review by brownleej — CHANGES_REQUESTED** [review]

File: internal/foundationdb_status.go (hunk 1 above)
Comment (brownleej) [reply internal/foundationdb_status.go:114]: I checked a 7.1 cluster with a large number of clients, and confirmed that the number of items in `max_protocol_clients` is smaller than the `max_protocol_count`, indicating that some of the clients are omitted.

Comment (johscheuer) [reply internal/foundationdb_status.go:114]: We discussed this offline and that's an expected behaviour. The list only contains a sample for the clients. This means we can't proceed with this PR as we might miss some real clients.

**Review by johscheuer — COMMENTED** [review]
I'm moving this PR into draft to signalling that we need some more information before proceeding.

**Comment by johscheuer** [comment]
Closed in favour of: https://github.com/apple/foundationdb/pull/9225

**Pull request closed without merging**